File: JAVA Codes/Roman_To_Int.java
import java.util.HashMap;
import java.util.Scanner;

public class Roman_To_Int {
    public int romanToInt(String s) {
        // Define a map for Roman numerals and their integer values
        HashMap<Character, Integer> romanMap = new HashMap<>();
        romanMap.put('I', 1);
        romanMap.put('V', 5);
        romanMap.put('X', 10);
        romanMap.put('L', 50);
        romanMap.put('C', 100);
        romanMap.put('D', 500);
        romanMap.put('M', 1000);

        int total = 0;
        int preValue = 0;

        // Traverse the string
        for (char c : s.toCharArray()) {
            int value = romanMap.get(c);
            if (value > preValue) {
                total += value - 2 * preValue;
            } else {
                total += value;
            }
            preValue = value;
        }

        return total;
    }

    public static void main(String[] args) {
        Roman_To_Int solution = new Roman_To_Int();
        Scanner sc = new Scanner(System.in);

        System.out.print("Enter a Roman numeral: ");
        String romanInput = sc.nextLine();

        int result = solution.romanToInt(romanInput);

        System.out.println("The integer value is: " + result);
    }
}

File: CPP Codes/RomanToInt.cpp
#include <iostream>
#include <string>
#include <unordered_map>
using namespace std;

class Solution 
{
public:

    int romanToInt(const string& s) 
    {
        
        unordered_map<char, int> romanMap = 
        {
            {'I', 1},
            {'V', 5},
            {'X', 10},
            {'L', 50},
            {'C', 100},
            {'D', 500},
            {'M', 1000}
        };

        int total = 0;
        int preValue = 0;

        for (char c : s) 
        {
            int value = romanMap[c];
            
            if (value > preValue) 
            {
                total += value - 2 * preValue;
            }
             else 
            {
                total += value;
            }
            preValue = value;
        }

        return total;
    }
};

int main()
{
    Solution solution;
    string romanInput;

    cout << "Enter a Roman numeral: ";
    cin >> romanInput;

    int result = solution.romanToInt(romanInput);

    cout << "The integer value is: " << result << endl;

    return 0;
}
